pkg/storage/elasticsearch.go

```diff
@@ -184,9 +166,22 @@ func (es ElasticSearch) GetEvents(ctx context.Context, filter *EventFilter, tena
 		}
 	}
 
-	offset := int(math.Min(float64(filter.Offset), float64(math.MaxInt32)))
-	limit := int(math.Min(float64(filter.Limit), float64(math.MaxInt32)))
-
+	// Explicitly clamp offset and limit to [0, MaxInt32] for offset, [1, MaxInt32] for limit
+	var offset int
+	if filter.Offset > uint(math.MaxInt32) {
+		offset = math.MaxInt32
+	} else {
+		offset = int(filter.Offset)
+	}
```

Comment:
```suggestion
    offset := int(filter.Offset)
	if offset > math.MaxInt32 {
		offset = math.MaxInt32
	}
```

Seems redundant to make two separate type assertions here.